### ISSUE TITLE
fix: 🤔 Combobox focus trap

### ISSUE THREAD

**Issue by auroraVasconcelos** [solid-design-system/solid#2734]
## Current behavior
When navigating through the combobox using the keyboard, the user gets trapped and it is not possible to exit the component.

## Expected behavior
When using keys such as Tab or, if necessary, Escape, the user should be able to exit the combobox and continue navigating through the page.

## Steps to reproduce
Go to the Combobox documentation page.
Use the keyboard to navigate through the component.

Try to exit the combobox using only the keyboard.

## DoR
- [ ] Item has business value
- [ ] Item has been estimated by the team
- [ ] Item is clear and well-defined
- [ ] Item dependencies have been identified
